apps/publish/enqueue/enqueue_service.py

```diff
@@ -555,6 +572,37 @@ def queue_transmission(self, doc, subscribers, subscriber_codes=None, associatio
         except Exception:
             raise
 
+    def remove_duplicate_association(self, associated_items):
+        associations = {}
+        for item in associated_items.values():
+            item_id = item.get("_id")
+            if item_id and item_id not in associations.keys():
+                associations[item_id] = item
+        return associations
+
+    def resend_association_items(self, doc):
+        associated_items = doc.get(ASSOCIATIONS)
+        if associated_items:
+            associated_items = self.remove_duplicate_association(associated_items)
```

Comment:
you're changing the structure here from `assoc-name: item` to `item-id: item` which is a bit error prone imo.
maybe something like `unique_associations = self.get_unique_associations(associated_items)` and return `associations.values()` from that method, you don't need the keys later

Reply:
yes, we don't need keys later so I will rename the `remove_duplicate_association` to `get_unique_associations` and return the `associations.values()`.